fpdf/svg.py

```diff
@@ -978,3 +984,89 @@ def apply_clipping_path(self, stylable, svg_element):
         if clipping_path:
             clipping_path_id = re.search(r"url\((\#\w+)\)", clipping_path)
             stylable.clipping_path = self.cross_references[clipping_path_id[1]]
+
+    @force_nodocument
+    def build_image(self, image):
+        if xmlns("xlink", "href") not in image.attrib:
+            raise ValueError("<image> is missing a href attribute")
+        width = float(image.attrib.get("width", 0))
+        height = float(image.attrib.get("height", 0))
+        if "preserveAspectRatio" in image.attrib:
```

Comment:
It might be nicer to only emit warnings for some of these cases. The result may not look like the user expected, but at least they get something.
Note that I have no idea how these attributes are actually used in practise, so this suggestion may or may not make sense (to be determined on a case-by-case basis).

Reply:
Yes, I considered this...

I prefer to be "strict" for now, and not allow those fields because:
* ignoring `transform` attributes will have a huge impact on the final layout of the vector graphics rendered
* we could make this behaviour configurable later one (with a "ignore_unsupported_svg_features" flag maybe?), but it's safer to switch from errors to warnings without bothering users, than the other way round (from warnings to errors)
* with visible errors like this, I hope that the `contributions are welcome!` message will attract contributors 😅

Reply:
It will be interesting to see which one of those we're going to be asked about first, since I really have no idea how often and in which ways they are used in practise.